src/Maestro/Maestro.Web/Api/v2020_02_20/Controllers/BuildsController.cs

```diff
@@ -30,14 +30,20 @@ namespace Maestro.Web.Api.v2020_02_20.Controllers
     public class BuildsController : v2019_01_16.Controllers.BuildsController
     {
         private IBackgroundQueue Queue { get; }
+        private IRemoteFactory _factory { get;  }
+        private readonly ILogger<BuildsController> _logger;
 
         public BuildsController(
             BuildAssetRegistryContext context,
             IBackgroundQueue queue,
-            ISystemClock clock)
+            ISystemClock clock,
+            IRemoteFactory factory,
```

Comment:
It's really sad to add the giant "IRemote" dependency to this class, as it will make it hard to test... could we put the commit method on some other controller?  It's weird because it's tying together two concepts.  The "commit" string has nothing to do with "Builds".

But I guess proper separation would require two web calls, which maybe isn't worth it?  And it's clear we don't have any testing of this class, since the tests didn't fail with this failed dependency. :-(

Reply:
I added this method in the build controller as on the frontend we use build.service to get the data (which is used in build.component).

Would you rather create a new controller "CommitController" then add a new service "commit.service" on the front-end side (which will be added in the constructor of build.component) @ChadNedzlek ? 

It seemed a little bit overkill to do this for the addition of 1 endpoint. But I completely agree that it can be confusing to have this new GetCommit endpoint inside the Build Controller.

Reply:
It's... probably fine.  I think the current problem is just that IRemote is such a huge interface, that testing is hard whenever it's involved... but that's not really the fault of this class.  And there is already an issue to break it up.

It's probably ok to merge this as is.